api/all/src/main/java/io/opentelemetry/api/metrics/Meter.java

```diff
@@ -8,71 +8,110 @@
+ *       <ul>
+ *         <li>If the value is monotonically increasing (the delta value is always non-negative) -
+ *             use a counter:
+ *             <pre>meter.counterBuilder("my-counter").build()</pre>
+ *         <li>If the value is NOT monotonically increasing (the delta value can be positive,
+ *             negative, or zero)) - use an up down counter:
+ *             <pre>meter.upDownCounterBuilder("my-up-down-counter").build()</pre>
+ *       </ul>
+ *   <li>I want to <b>record</b> or <b>time</b> something, and the <b>statistics</b> about this
+ *       thing are likely to be meaningful - use a histogram:
+ *       <pre>meter.histogramBuilder("my-histogram").build()</pre>
+ *   <li>I want to <b>measure</b> something (by reporting an absolute value):
  *       <ul>
- *         <li>The value is always increasing / I want to track its <b>rate</b>.<br>
- *             Use {@link #counterBuilder(String)}
- *         <li>The value is not always increasing.<br>
- *             Use {@link #upDownCounterBuilder(String)}
+ *         <li>If it makes NO sense to add up the values across different sets of attributes, use an
+ *             asynchronous gauge:
+ *             <pre>
+ *             meter.gaugeBuilder("my-gauge").buildWithCallback(observableMeasurement -> observableMeasurement.record(..))
+ *             </pre>
+ *         <li>If it makes sense to add up the values across different sets of attributes:
+ *             <ul>
+ *               <li>If the value is monotonically increasing - use an asynchronous counter:
+ *                   <pre>
+ *                   meter.counterBuilder("my-async-counter").buildWithCallback(observableMeasurement -> observableMeasurement.record(..))
+ *                   </pre>
+ *               <li>If the value is NOT monotonically increasing - use an asynchronous up down
+ *                   counter:
+ *                   <pre>
+ *                   meter.upDownCounterBuilder("my-async-counter").buildWithCallback(observableMeasurement -> observableMeasurement.record(..))
+ *                   </pre>
+ *             </ul>
  *       </ul>
- *   <li>I want to <b>time</b> something, or record measurements where the statistics are important
- *       (e.g. latency).<br>
- *       <b>Use {@link #histogramBuilder(String)}</b>
- *   <li>I want to measure something by sampling a value stored elsewhere. <br>
- *       Use {@link #gaugeBuilder(String)}
  * </ul>
+ *
+ * @see <a
+ *     href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/supplementary-guidelines.md#instrument-selection">Instrument
+ *     Selection Guidelines</a>
  */
 @ThreadSafe
 public interface Meter {
   /**
    * Constructs a counter instrument.
    *
-   * <p>This is used to build both synchronous (in-context) instruments and asynchronous (callback)
-   * instruments.
+   * <p>This is used to build both synchronous instruments and asynchronous instruments (i.e.
+   * callbacks).
```

Comment:
```suggestion
   * <p>This is used to build both synchronous (in-context) instruments and asynchronous instruments (i.e.
   * callbacks).
```
Minor: I like the original inclusion of "in-context" because I think it's an important aspect of the difference between the two. The spec notes the distinction about context too.

Reply:
Measurements _can_ be recorded in context for synchronous instruments but don't need to be. I omitted this because this method is simply the entry point for a builder - it needs to make clear that the builder builds both synchronous and async varieties, but that's it. I think the "i.e. callbacks" clause for asynchronous instruments is different because the method name refers to callback `buildWithCallback`, so we need to make clear that callback = asynchronous.

Reply:
Oh yeah good point about "in-context" not being being a given, just a possibility. I hadn't thought about it suggesting such...but you're totally right. Looks good with your change.